src/main/java/com/cleanroommc/groovyscript/compat/mods/theaurorian/MoonlightForge.java

```diff
@@ -0,0 +1,106 @@
+        restoreFromBackup().forEach(MoonlightForgeRecipeHandler::addRecipe);
+        removeScripted().forEach(r -> MoonlightForgeRecipeHandler.allRecipes.removeIf(u -> u.equals(r)));
+    }
+
+    public void add(MoonlightForgeRecipe recipe) {
+        addScripted(recipe);
+        MoonlightForgeRecipeHandler.addRecipe(recipe);
+    }
+
+    public boolean remove(MoonlightForgeRecipe recipe) {
+        return MoonlightForgeRecipeHandler.allRecipes.removeIf(r -> {
+            if (r.equals(recipe)) {
+                addBackup(recipe);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(example = @Example("item('theaurorian:aurorianiteingot')"))
+    public boolean removeByInput(IIngredient output) {
+        return MoonlightForgeRecipeHandler.allRecipes.removeIf(r -> {
+            if (output.test(r.getInput1()) || output.test(r.getInput2())) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(example = @Example("item('theaurorian:queenschipper')"))
+    public boolean removeByOutput(IIngredient output) {
+        return MoonlightForgeRecipeHandler.allRecipes.removeIf(r -> {
+            if (output.test(r.getOutput())) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        MoonlightForgeRecipeHandler.allRecipes.forEach(this::addBackup);
+        MoonlightForgeRecipeHandler.allRecipes.clear();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<MoonlightForgeRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(MoonlightForgeRecipeHandler.allRecipes).setRemover(this::remove);
+    }
+
+    @Property(property = "input", valid = @Comp("2"))
+    @Property(property = "output", valid = @Comp("1"))
+    public static class RecipeBuilder extends AbstractRecipeBuilder<MoonlightForgeRecipe> {
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Moonlight Forge recipe";
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 2, 2, 1, 1);
+            validateFluids(msg);
+        }
+
+        @Override
+        @RecipeBuilderRegistrationMethod
+        public @Nullable MoonlightForgeRecipe register() {
+            if (!validate()) return null;
+            MoonlightForgeRecipe recipe = null;
+            for (ItemStack input1 : input.get(0).getMatchingStacks()) {
+                for (ItemStack input2 : input.get(1).getMatchingStacks()) {
+                    recipe = new MoonlightForgeRecipe(input1, input2, output.get(0));
+                    ModSupport.THE_AURORIAN.get().moonlightForge.add(recipe);
+                }
+            }

```

Comment:
I really hate this. Might be better to force item stacks

Reply:
I mean this is not ideal but I don't see a reason why not?

Reply:
If both inputs are oredicts with 10 entries each (which is rare, but can happen), then it will add 100 recipes. But its probably fine most of the time.

Reply:
Well that's kind of on the modpack dev that they did this? Not to mention if they don't use unidict then that's just something they might have to do anyway